src/app/client/src/app/app.component.ts

```diff
@@ -122,7 +123,42 @@ export class AppComponent implements OnInit, OnDestroy {
           _.get(this.activatedRoute, 'snapshot.firstChild.firstChild.firstChild.data.hideHeaderNFooter');
       });
   }
+  public makeUTMSession(params) {
+    for (const item in params) {
+      if (params.hasOwnProperty(item)) {
+        switch (item) {
+          case ('channel'):
+            this.resultJson.push({ 'id': params[item], 'type': 'Source' });
+            break;
+          case ('utm_campaign'):
+            this.resultJson.push({ 'id': params[item], 'type': 'Source' });
+            break;
+          case 'utm_medium':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmMedium' });
+            break;
+          case 'utm_source':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmSource' });
+            break;
+          case 'utm_term':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmTerm' });
+            break;
+          case 'utm_content':
+            this.resultJson.push({ 'id': params[item], 'type': 'UtmContent' });
+            break;
+          default:
+            break;
+        }
+      }
+    }
+    sessionStorage.setItem('UTM', JSON.stringify(this.resultJson));
```

Comment:
why are using sessionStorage directly. we have service for this right

Reply:
I had discussed with rajeev he told me why you are using service for this so I am using session storage.

Reply:
hey If I had said that I am really sorry we should use the service for session store please don't use directly

Reply:
Done with storing inside telemetry service and making 1 copy to session storage also.